File: app/src/main/java/com/watnapp/etipitaka/plus/model/Favorite.java
package com.watnapp.etipitaka.plus.model;
import android.provider.BaseColumns;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import com.watnapp.etipitaka.plus.helper.BookDatabaseHelper;

public class Favorite extends ModelBase {
  private int id;
  private String note;
  private int language;
  private int volume;
  private int page;
  private int item;
  private int score;


  public Favorite() {
    super();
  }

  @Override
  public int getId() {
    return id;
  }

  @Override
  public void setId(int id) {
    this.id = id;
  }

  public String getNote() {
    return note;
  }

  public void setNote(String note) {
    this.note = note;
  }

  public BookDatabaseHelper.Language getLanguage() {
    return BookDatabaseHelper.Language.values()[language];
  }

  public void setLanguage(BookDatabaseHelper.Language language) {
    this.language = language.ordinal();
  }

  public int getVolume() {
    return volume;
  }

  public void setVolume(int volume) {
    this.volume = volume;
  }

  public int getPage() {
    return page;
  }

  public void setPage(int page) {
    this.page = page;
  }

  public int getItem() {
    return item;
  }

  public void setItem(int item) {
    this.item = item;
  }

  public int getScore() {
    return score;
  }

  public void setScore(int score) {
    this.score = score;
  }

  @Override
  public void fromCursor(Cursor cursor, Context context) {
    this.id = cursor.getInt(cursor.getColumnIndexOrThrow(BaseColumns._ID));
    this.note = cursor.getString(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.NOTE));
    this.language = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.LANGUAGE));
    this.volume = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.VOLUME));
    this.page = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.PAGE));
    this.item = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.ITEM));
    this.score = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.SCORE));
  }

  @Override
  public ContentValues toContentValues() {
    ContentValues values = new ContentValues();
    values.put(FavoriteTable.FavoriteColumns.NOTE, this.note);
    values.put(FavoriteTable.FavoriteColumns.LANGUAGE, this.language);
    values.put(FavoriteTable.FavoriteColumns.VOLUME, this.volume);
    values.put(FavoriteTable.FavoriteColumns.PAGE, this.page);
    values.put(FavoriteTable.FavoriteColumns.ITEM, this.item);
    values.put(FavoriteTable.FavoriteColumns.SCORE, this.score);
    return values;
  }

  public static Favorite newInstance(Cursor cursor, Context context) {
    Favorite favorite = new Favorite();
    favorite.fromCursor(cursor, context);
    return favorite;
  }

}

File: app/src/main/java/com/watnapp/etipitaka/plus/model/Favorite.kt
package com.watnapp.etipitaka.plus.model

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.provider.BaseColumns
import com.watnapp.etipitaka.plus.helper.BookDatabaseHelper

class Favorite : ModelBase() {
    private var id = 0
    private var note: String? = null
    private var language = 0
    private var volume = 0
    private var page = 0
    private var item = 0
    private var score = 0

    override fun getId(): Int = id

    override fun setId(id: Int) {
        this.id = id
    }

    fun getNote(): String? = note

    fun setNote(note: String?) {
        this.note = note
    }

    fun getLanguage(): BookDatabaseHelper.Language = BookDatabaseHelper.Language.values()[language]

    fun setLanguage(language: BookDatabaseHelper.Language) {
        this.language = language.ordinal
    }

    fun getVolume(): Int = volume

    fun setVolume(volume: Int) {
        this.volume = volume
    }

    fun getPage(): Int = page

    fun setPage(page: Int) {
        this.page = page
    }

    fun getItem(): Int = item

    fun setItem(item: Int) {
        this.item = item
    }

    fun getScore(): Int = score

    fun setScore(score: Int) {
        this.score = score
    }

    override fun fromCursor(cursor: Cursor, context: Context) {
        id = cursor.getInt(cursor.getColumnIndexOrThrow(BaseColumns._ID))
        note = cursor.getString(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.NOTE))
        language = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.LANGUAGE))
        volume = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.VOLUME))
        page = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.PAGE))
        item = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.ITEM))
        score = cursor.getInt(cursor.getColumnIndexOrThrow(FavoriteTable.FavoriteColumns.SCORE))
    }

    override fun toContentValues(): ContentValues =
        ContentValues().apply {
            put(FavoriteTable.FavoriteColumns.NOTE, note)
            put(FavoriteTable.FavoriteColumns.LANGUAGE, language)
            put(FavoriteTable.FavoriteColumns.VOLUME, volume)
            put(FavoriteTable.FavoriteColumns.PAGE, page)
            put(FavoriteTable.FavoriteColumns.ITEM, item)
            put(FavoriteTable.FavoriteColumns.SCORE, score)
        }

    companion object {
        @JvmStatic
        fun newInstance(cursor: Cursor, context: Context): Favorite =
            Favorite().apply { fromCursor(cursor, context) }
    }
}
